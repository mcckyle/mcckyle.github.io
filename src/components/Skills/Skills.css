/* Filename: Skills.css */

/* =======================
   Skills Styles.
========================== */
#skills .section-title
{
    color: var(--secondary);
}

#skills .card
{
    border-left: 4px solid var(--highlight);
}

.skills-list
{
    display: grid;
    gap: 1.75rem;
    padding: 0;
    margin: 0;
    list-style: none;
}

.skills-subtitle
{
    font-size: 1.25rem;
    font-weight: 600;
    color: var(--primary);
    margin-bottom: 0.5rem;
}
