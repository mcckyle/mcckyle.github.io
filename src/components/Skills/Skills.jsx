//Filename: Skills.jsx
//Author: Kyle McColgan
//Date: 3 October 2025
//Description: This file contains the component for the personal React project skills section.

import React from "react";
import "./Skills.css";

function Skills()
{
  const skills = [
    {
      title: "Application Development",
      description:
        "Proficient in Java and JavaScript, building scalable web apps with clean architecture. Experienced with React on the front end and Spring Boot on the backend to deliver responsive full-stack solutions.",
    },
    {
      title: "Frontend Engineering",
      description:
        "Skilled in HTML, CSS, and modern JavaScript. I use React to create intuitive, accessible UIs and incorporate libraries like Bootstrap and Material-UI to accelerate design implementation.",
    },
    {
      title: "Backend & Systems",
      description:
        "Comfortable working with Java, C++, and Python for backend development and systems programming. I focus on writing efficient, secure, and well-documented code for maintainability and performance.",
    },
    {
      title: "Information Security",
      description:
        "Solid understanding of secure coding practices, data protection, and threat mitigation. Actively applying these principles in personal projects to reinforce privacy, integrity, and robust application design.",
    },
  ];

  return (
    <section id="skills" aria-labelledby="skills-title" className="section">
      <h2 id="skills-title" className="section-title">
        Skills
      </h2>
      <ul className="skills-list">
      {skills.map((skill, index) => (
        <li key={index} className="card skills-card">
          <h3 className="skills-subtitle">{skill.title}</h3>
          <p>{skill.description}</p>
        </li>
      ))}
      </ul>
    </section>
  );
}

export default Skills;
