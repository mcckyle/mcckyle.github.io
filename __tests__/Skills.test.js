//Filename: Skills.test.js
//Author: Kyle McColgan
//Date: 29 September 2025
//Description: This file contains the Jest unit tests for the React personal project skills section.

import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Skills from '../src/components/Skills/Skills.jsx';

it('renders the Skills section', () => {
    render(<Skills />);
    const section = screen.getByRole('region', { name: /Skills/i });
    expect(section).toBeInTheDocument();
});

it('renders the Skills heading', () => {
    render(<Skills />);
    const heading = screen.getByRole('heading', { level: 2 });
    expect(heading).toHaveTextContent('Skills');
});

it('renders four skill list items', () => {
    render(<Skills />);
    const listItems = screen.getAllByRole('listitem');
    expect(listItems).toHaveLength(4);
});

it('renders the Application Development skill category', () => {
    render(<Skills />);
    const category = screen.getByText(/Application Development:/);
    expect(category).toBeInTheDocument();
});

it('renders the Frontend Engineering skill category', () => {
    render(<Skills />);
    const category = screen.getByText(/Frontend Engineering:/);
    expect(category).toBeInTheDocument();
});

// it('renders all skill categories in bold', () => {
//     render(<Skills />);
//     const boldElements = screen.getAllByRole('strong');
//     expect(boldElements).toHaveLength(4);
// });

it('renders the skills list as a <ul>', () => {
    render(<Skills />);
    const list = screen.getByRole('list');
    expect(list.tagName).toBe('UL');
});

it('does not render any empty <li> elements', () => {
    render(<Skills />);
    const items = screen.getAllByRole('listitem');
    items.forEach(item => expect(item).not.toBeEmptyDOMElement());
});

it('matches the snapshot of the Skills section', () => {
    const { asFragment } = render(<Skills />);
    expect(asFragment()).toMatchSnapshot();
});

it('renders without console errors', () => {
    const consoleError = jest.spyOn(console, 'error').mockImplementation(() => {});
    render(<Skills />);
    expect(consoleError).not.toHaveBeenCalled();
    consoleError.mockRestore();
});
