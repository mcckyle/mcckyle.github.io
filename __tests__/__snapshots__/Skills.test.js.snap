// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills Component matches the snapshot of the Skills section 1`] = `
<DocumentFragment>
  <section
    aria-labelledby="skills-title"
    class="section"
    id="skills"
  >
    <h2
      class="section-title"
      id="skills-title"
    >
      Skills
    </h2>
    <ul
      class="skills-list"
    >
      <li
        class="card skills-card"
      >
        <h3
          class="skills-subtitle"
        >
          Application Development
        </h3>
        <p>
          Proficient in Java and JavaScript, building scalable web apps with clean architecture. Experienced with React on the front end and Spring Boot on the backend to deliver responsive full-stack solutions.
        </p>
      </li>
      <li
        class="card skills-card"
      >
        <h3
          class="skills-subtitle"
        >
          Frontend Engineering
        </h3>
        <p>
          Skilled in HTML, CSS, and modern JavaScript. I use React to create intuitive, accessible UIs and incorporate libraries like Bootstrap and Material-UI to accelerate design implementation.
        </p>
      </li>
      <li
        class="card skills-card"
      >
        <h3
          class="skills-subtitle"
        >
          Backend & Systems
        </h3>
        <p>
          Comfortable working with Java, C++, and Python for backend development and systems programming. I focus on writing efficient, secure, and well-documented code for maintainability and performance.
        </p>
      </li>
      <li
        class="card skills-card"
      >
        <h3
          class="skills-subtitle"
        >
          Information Security
        </h3>
        <p>
          Solid understanding of secure coding practices, data protection, and threat mitigation. Actively applying these principles in personal projects to reinforce privacy, integrity, and robust application design.
        </p>
      </li>
    </ul>
  </section>
</DocumentFragment>
`;
